app/src/main/java/com/aaronjwood/portauthority/network/Host.java

```diff
@@ -53,6 +53,9 @@ public String getHostname() {
      * @param hostname Hostname for this host
      */
     public void setHostname(String hostname) {
+        if (hostname != null && hostname.isEmpty() && hostname.endsWith(".local")) {
```

Comment:
`hostname.isEmpty() && hostname.endsWith(".local")` seems like it should be impossible.